graphviz/jupyter_integration.py

```diff
@@ -1,12 +1,85 @@
 """Display rendered graph as SVG in Jupyter Notebooks and QtConsole."""
-
 from . import piping
 
-__all__ = ['JupyterSvgIntegration']
+__all__ = ['JUPYTER_REPRESENTATIONS',
+           'verify_jupyter_representation',
+           'JupyterSvgIntegration']
+
+JUPYTER_REPRESENTATIONS = {
+    'image/svg+xml': '_repr_image_svg_xml',
+    'image/png': '_repr_image_png',
+    'image/jpeg': '_repr_image_jpeg',
+
+    # The following does not work yet. Not clear why, because the documentation
+    # has only few information about them.
+    'application/json': None,  # ToDo: What does jupyter expect?
+    'application/pdf': None,  # ToDo: Does Jupyter only accept file path?
+
+    # No reasonable representation
+    'text/plain': None,
+    'text/latex': None,
+    'text/html': None,
+    'application/javascript': None,
+    'text/markdown': None,
+}
+
+DEFAULT_JUPYTER_REPRESENTATION = 'image/svg+xml'
+
+REQUIRED = True
+
+
+def verify_jupyter_representation(jupyter_representation: str,
+                                  *,
+                                  required: bool = REQUIRED) -> None:
+    if jupyter_representation is None:
+        if required:
+            raise ValueError('missing jupyter_representation')
+    elif jupyter_representation not in JUPYTER_REPRESENTATIONS:
+        raise ValueError(
+            f'unknown jupyter_representation: {jupyter_representation!r}')
 
 
 class JupyterSvgIntegration(piping.Pipe):
     """Display rendered graph as SVG in Jupyter Notebooks and QtConsole."""
 
-    def _repr_svg_(self):
+    _jupyter_representation = DEFAULT_JUPYTER_REPRESENTATION
+
+    _verify_jupyter_representation = staticmethod(verify_jupyter_representation)
+
+    @property
+    def jupyter_representation(self) -> str:
```

Comment:
Do you think it is essential to set the output format also per-instance as opposed to setting a module-default that applies to all instances?

Reply:
No, I copied the tests, and they wanted this.
Since changing instance values has less side effects than module-defaults, I just added this.

Reply:
I see. I am on the fence (maybe I would call it `jupyter_format` if we want to expose it per-instance). Might adapt the tests then,

Reply:
`jupyter_repr` could also be used. While `repr` is a shortcut, it is common and each Python user should know it. Further the name is close to `_repr_svg_`.